### PR TITLE
Support globbing in directive list arguments

### PR DESCRIPTION
**Pull request by dapomeroy**
Adds support for wildcards in list arguments within directives, so that arbitrary workloads can be supported more easily.

### CODE REVIEW

File: lib/ramble/ramble/test/end_to_end/globbing_patterns.py
Comment (dapomeroy) [inline lib/ramble/ramble/test/end_to_end/globbing_patterns.py:1]: What's the right amount of rigor to add to the tests here? This one checks workload(executables), and all the rest use the same helper method for expansion. Seems to run quite fast. Think it's worthwhile to add tests for variables, env_vars, mods, etc.?

Comment (douglasjacobsen) [reply lib/ramble/ramble/test/end_to_end/globbing_patterns.py:1]: Yeah, I think we should try to use a kind of ridiculous amount of globbing to help make sure it all works.

If it's in an end-to-end test, you can just test all of it in a single test though.

File: lib/ramble/ramble/language/application_language.py [hunk 1]
```diff
@@ -71,13 +71,22 @@ def workload(name, executables=None, executable=None, input=None,
     """
 
     def _execute_workload(app):
+        exp_executables = executables
+        if isinstance(exp_executables, list):
+            exp_executables = ramble.language.language_helpers.expand_patterns(app.executables,
```
Comment (douglasjacobsen) [inline lib/ramble/ramble/language/application_language.py]: Could this logic be pushed down into `require_definition` or `merge_definition`? (i.e. the list check that then expands and merges with the singular definition)

File: lib/ramble/ramble/language/modifier_language.py [hunk 2]
```diff
@@ -81,8 +81,13 @@ def _execute_variable_modification(mod):
             raise DirectiveError('variable_modification directive given an invalid method.\n'
                                  f'  Valid methods are {str(supported_methods)}')
 
+        exp_modes = modes
+        if isinstance(exp_modes, list):
```
Comment (douglasjacobsen) [inline lib/ramble/ramble/language/modifier_language.py]: Same comment about pushing the logic down.

**Review by douglasjacobsen — CHANGES_REQUESTED** [review]
One request. Otherwise LGTM! Thanks!

File: lib/ramble/ramble/language/application_language.py (hunk 1 above)
Comment (dapomeroy) [reply lib/ramble/ramble/language/application_language.py]: Definitely. I've updated this to be in `merge_definition` and it's much cleaner now.

File: lib/ramble/ramble/language/modifier_language.py (hunk 2 above)
Comment (dapomeroy) [reply lib/ramble/ramble/language/modifier_language.py]: Done

File: lib/ramble/ramble/test/end_to_end/globbing_patterns.py
Comment (dapomeroy) [reply lib/ramble/ramble/test/end_to_end/globbing_patterns.py:1]: OK, I've added a lot to the test and it still runs fast.

**Review by douglasjacobsen — APPROVED** [review]

**Pull request merged**